test/winners.js

```diff
@@ -0,0 +1,15 @@
+const tap = require('tap')
+const _ = require('underscore')
+
+const {Synopsis} = require('../index.js')
+
+const weights = [
+  { publisher: '10%', weight: 0.10, pinPercentage: 10 },
+  { publisher: '20%', weight: 0.20, pinPercentage: 20 },
+  { publisher: '30%', weight: 0.30, pinPercentage: 30 },
+  { publisher: '25%', weight: 0.25, pinPercentage: 25 },
+  { publisher: '15%', weight: 0.15, pinPercentage: 15 }
+]
```

Comment:
would be great to also test the case where `n = 0`  and the case with `pinPercentage: 0`

Reply:
also this might be a dumb q, but what happens if every entry in weights has a pinPercentage, but the pinPercentages don't add up to 100?

Reply:
My understanding of the pinning model is that the sum of the pinPercentages is the fraction of the votes you want to go to pinned sites; the rest of the votes can go to any (pinned or unpinned) sites according to attention (duration, #visits) spent on them.

Reply:
`pinPercentage: 0` case added in 765e8112b9f6f561ef2fb824e774691dc33e57f4.

Turns out `n = 0` case yields a single result.  Didn't expect that!